### PR TITLE
Avoid exception during compliation

### PR DESCRIPTION
**Pull request by mdesmet**
## Overview

Avoid exception during compilation

## Checklist

- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] `README.md` updated and added information about my change
- [ ] `CHANGELOG.md` updated and added information about my change

### CODE REVIEW

File: dbt/include/trino/macros/crossdb/datediff.sql [hunk 1]
```diff
@@ -23,6 +23,6 @@
     {% elif datepart == 'millisecond' %}
         (to_milliseconds((CAST({{ second_date }} AS TIMESTAMP) - CAST({{ first_date }} AS TIMESTAMP))))
     {% else %}
-        {{ exceptions.raise_compiler_error("Unsupported datepart for macro datediff in Trino: {!r}".format(datepart)) }}
+        {% if execute %}{{ exceptions.raise_compiler_error("Unsupported datepart for macro datediff in Trino: {!r}".format(datepart)) }}{% endif %}
```
Comment (findinpath) [inline dbt/include/trino/macros/crossdb/datediff.sql:26]: what problem were you trying to solve?

Comment (findinpath) [reply dbt/include/trino/macros/crossdb/datediff.sql:26]: I found in the meantime: 
https://github.com/dbt-labs/dbt-core/blob/a7ff003d4f74763f126e77a0fc9c55e7a3311b3d/core/dbt/context/providers.py#L914-L970

Comment (findinpath) [inline dbt/include/trino/macros/crossdb/datediff.sql:26]: nit: could you modify to 

```
{% if execute %}
{{ exceptions.raise_compiler_error("Unsupported datepart for macro datediff in Trino: {!r}".format(datepart)) }}
{% endif %}
```

as suggested in the code snippet referenced above?

**Comment by findinpath** [comment]
Do we need a new minor release or was this already included in `1.2.0` ?

**Comment by mdesmet** [comment]
This is only impacting the trino_dbt_utils integration tests as the standard trino_dbt_utils tests are applying an unsupported datepart (`microseconds` is not supported in Trino right now). It seems that dbt even when disabling a model, still compiles the disabled model and fails. It should not impact any projects using dbt-trino 1.2.

**Pull request merged**